pkg/plugins/awsami/spec.go

```diff
@@ -50,10 +50,6 @@ func (s *Spec) Validate() (errs []error) {
 		s.Endpoint = fmt.Sprintf("https://ec2.%s.amazonaws.com", s.Region)
 	}
 
-	if len(s.Filters) == 0 {
```

Comment:
Can you elaborate on the motivation to remove this validation?

Reply:
The validation is unaware if the resource is a source or a condition, and the validation is run during the object initialization.

In the case of a condition, the list of filter can definitively be empty.

But it means that I forgot to set this validation at "runtime" in source.go though. Let me fix that.

Reply:
@olblak fixed by adding the validation at runtime for source. 
Long term fix will be to have the resource's Validation function to be aware of the type (source/condition/target) and the context (scm, etc.) to decide what to validate and what NOT to validate.